src/hatch/utils/auth.py

```diff
@@ -0,0 +1,117 @@
+from __future__ import annotations
+
+from hatch.utils.fs import Path
+
+
+class AuthenticationCredentials:
```

Comment:
Let's put this in `src/hatch/publish/index.py` or make index a package and put this in `src/hatch/publish/index/auth.py`